Testing/gamecontroller/GameController.gd

```diff
@@ -0,0 +1,125 @@
+extends Node
+
+export var first_level := "first_level"
+export var main_menu_scene := ""
+
+#current scene being played. E.g.: First level
+var current_scene = null
+
+#Player status scene file name
+export var player_status := ""
+#Player status node
+var player_status_node = null
+#reference to player node
+var player_node = null
+
+onready var loading_screen := $Menu/LoadingScreen
+onready var pause_menu_screen := $Menu/PauseMenu
+onready var game_over_screen := $Menu/GameOverScreen
+
+enum GameState {MAINMENU, PAUSEMENU, PLAYING, EXITING, GAMEOVER, LOADING}
+var state = GameState.PLAYING
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	create_player_status ()
+	if loading_screen != null:
+		loading_screen.hide()
+	if pause_menu_screen != null:
+		pause_menu_screen.hide()
+	if game_over_screen != null:
+		game_over_screen.hide()
+	
+	var root = get_tree().get_root()
+	current_scene = root.get_child(root.get_child_count() - 1)
+
+func create_player_status () -> void:
+	if player_status_node == null:
+		var status = ResourceLoader.load(player_status)
+		player_status_node = status.instance()
+		add_child(player_status_node)
+
+func start_game () -> void:
+	call_deferred("_load_scene", first_level)
+	state = GameState.PLAYING
+
+func load_main_menu () -> void:
+	load_level(main_menu_scene)
+	resume_paused ()
+	state = GameState.MAINMENU
+
+func load_level (scenepath) -> void:
```

Comment:
maybe safe the game here? prob only while state == GameState.PLAYING

Reply:
Maybe, I though we were only going to save on the checkpoints. Should I add it?

Reply:
we could also have an autosafe trigger on every level start. maybe we could also encode the initial spawn possition as a checkpoint. 

It just felt a bit strange loading the game and ending up in the last level